ir/ir.def

```diff
@@ -81,25 +81,20 @@ class ParserState : ISimpleNamespace, Declaration, IAnnotated {
 }
 
 // A parser that contains all states (unlike the P4 v1.0 parser, which is really just a state)
-class P4Parser : Type_Declaration, ISimpleNamespace, IApply, IContainer {
+class P4Parser : Type_Declaration, INestedNamespace, ISimpleNamespace, IApply, IContainer {
```

Comment:
Are these the only two nested namespaces?

Reply:
These are the only two objects that have both arguments and locals that might shadow the arguments, so need this special handling to get the P4_16 name lookup correct.  Function and Action objects don't have any locals directly (any locals will be nested in the body BlockStatement)

Reply:
Actually, I take that back -- extern type might also, so I should extend this to them (though it can't happen in a user program, as users can't define extern types -- only architectures)